common/djangoapps/philu_commands/management/commands/send_referral_follow_up_email.py

```diff
@@ -0,0 +1,41 @@
+from datetime import timedelta
+from django.utils import timezone
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+
+from openedx.features.smart_referral.models import SmartReferral
+from openedx.features.smart_referral.tasks import task_send_referral_follow_up_email
+
+CONTACT_EMAIL = 'contact_email'
+DAYS_TO_SEND_FOLLOW_UP_EMAIL = 3
+IS_REFERRAL_STEPS_COMPLETED = 'is_referral_step_complete'
+
+
+class Command(BaseCommand):
+    help = """
+    Send follow up referral email to all those contacts
+    """
+
+    def handle(self, *args, **options):
+        emails_list = []
+        three_days_old_date = timezone.now().date() - timedelta(days=DAYS_TO_SEND_FOLLOW_UP_EMAIL)
+
+        three_days_old_referral = SmartReferral.objects.filter(
+            created__date__lte=three_days_old_date,
+            is_referral_step_complete=False
+        ).values(
+            CONTACT_EMAIL,
+            IS_REFERRAL_STEPS_COMPLETED
+        )
+        for referral in three_days_old_referral:
+            try:
+                User.objects.get(email=referral[CONTACT_EMAIL])
+                SmartReferral.objects.filter(contact_email=referral[CONTACT_EMAIL]).update(
+                    is_referral_step_complete=True
+                )
```

Comment:
We can have a custom `ModelManager` for `SmartReferral` model and add a method in this manager. 
```
def complete_referral_step(self, contact_email): 
      SmartReferral.objects.filter(contact_email=contact_email).update(is_referral_step_complete=True)
```

And it can be reused here and in task as well.
```
SmartReferral.objects.complete_referral_step(contact_email=referral[CONTACT_EMAIL])
```

Reply:
It's a good suggestion, but I think it will be overkill at this point. Thanks though.